service/src/Agg.cpp

```diff
@@ -66,6 +66,22 @@ namespace geopm
         return result;
     }
 
+    double Agg::bitwise_or(const std::vector<double> &operand)
+    {
+        auto filtered = nan_filter(operand);
+        double result = NAN;
+        uint64_t agg_tmp = 0;
+        if (filtered.size()) {
+            for (const auto &it : filtered) {
+                if (it >= 0) {
```

Comment:
Lets not support -1 as a special value in the this implementation, rather do it on the LevelZeroIOGroup side.